jax/_src/lru_cache.py

```diff
@@ -153,26 +171,36 @@ def _evict_if_needed(self, *, additional_size: int = 0) -> None:
     if not self.eviction_enabled:
       return
 
-    # a priority queue, each element is a tuple `(file_mtime, file, file_size)`
-    h: list[tuple[int, pathlib.Path, int]] = []
+    # a priority queue, each element is a tuple `(file_atime, key, file_size)`
+    h: list[tuple[int, str, int]] = []
     dir_size = 0
-    for file in self.path.iterdir():
-      if file.is_file():
-        file_size = file.stat().st_size
-        file_mtime = file.stat().st_mtime_ns
+    for cache_file in self.path.glob(f"*{self._cache_suffix}"):
+      if cache_file.is_file():
```

Comment:
Flip the condition and `continue` to reduce the nesting level:

```
if not cache_file.is_file():
  continue
```

Reply:
Done